src/style/grid.rs

```diff
@@ -0,0 +1,446 @@
+//! Style types for CSS Grid layout
+use super::{AlignContent, LengthPercentage, Style};
+use crate::axis::{AbsoluteAxis, AbstractAxis};
+use crate::geometry::{Line, MinMax};
+use crate::style::AvailableSpace;
+use crate::style_helpers::*;
+use crate::sys::GridTrackVec;
+use core::cmp::{max, min};
+
+/// Controls whether grid items are placed row-wise or column-wise. And whether the sparse or dense packing algorithm is used.
+/// The "dense" packing algorithm attempts to fill in holes earlier in the grid, if smaller items come up later. This may cause items to appear out-of-order, when doing so would fill in holes left by larger items.
```

Comment:
```suggestion
///
/// The "dense" packing algorithm attempts to fill in holes earlier in the grid, if smaller items come up later. This may cause items to appear out-of-order, when doing so would fill in holes left by larger items.
```

Line breaks will lead to nicer tooltips in IDEs.